averaging/disc_aver_sw_paradiag_with_switches.py

```diff
@@ -186,20 +192,23 @@
         'atol': atol,
         'stol': 1e-12,
     },
-    'pc_type': 'python',
-    'pc_python_type': 'asQ.CirculantPC',
-    'circulant_alpha': args.alphap,
-    'circulant_state': 'linear',
-    'aaos_jacobian_state': 'linear',
+    'pc_type': 'composite',
+    'pc_composite_pcs': 'python,python',
+    'pc_composite_type': 'multiplicative',
+    'sub_0' : {'pc_python_type': 'asQ.CirculantPC',
+               'circulant_alpha': args.alphap,
+               'circulant_state': 'linear',
+               'aaos_jacobian_state': 'linear'},
+    'sub_1' : {'pc_python_type': 'asQ.JacobiPC',
+               'aaojacobi_block': block_parameters,
+               'aaojacobi_state': 'linear',
+               'aaos_jacobian_state': 'linear'}
 }
 
 # setup parameters for paradiag solve (the number of windows is fixed to 1)
-if args.advection:
-    solver_parameters_diag['circulant_block_'+str(i)+'_'] = monoparameters_ns
-    print('set monoparameters_ns as solver_parameters_diag')
-else:
-    solver_parameters_diag['circulant_block_'+str(i)+'_'] = hparams
-    print('set hparams as solver_parameters_diag')
+for i in range(sum(time_partition_t)):
+    solver_parameters_diag[
+        'sub_0_circulant_block_'+str(i)+'_'] = block_parameters
```

Comment:
You can now just add `'circulant_block': block_parameters` to the dictionary above and it will apply those params to all blocks. This is better for very large Nt because the options dictionary is much smaller